lec17-rmarkdown.Rmd

```diff
@@ -1,4 +1,306 @@
 ---
-title: "Reproducible workflow, metadata, R Markdown"
+title: "Reproducible workflow, R Markdown"
 ---
+## Lesson preamble
 
+> ### Lesson objectives:
+>
+> - Learn why reproducibility in science is important
+> - Discuss how we can improve reproducibility of our research
+> - RMarkdown formatting: including images, citations, footnotes, and output formats.
+>
+> ### Lesson outline:
+>
+> - Reproducible Science (50 min)
+>     - Introduction to the problem and discussion (20 min)
+>     - How we can improve reproducibility (15 min)
+>     - What are the barriers to reproducibility? (15 min)
+> - Intermediate topics in R Markdown (50 min)
+>     - Online Tutorial (10 min)
+>     - Lists, Tables
+>     - Images, Figures
+>     - In-Line Citations & Bibliography
+>     - Footnotes
+>     - Output Formatting Options
+
+----
+## Reproducible Science
+
+What does reproducibility mean to you? Let's discuss what each of these may entail:
+
+- Computational reproducibility
+- Scientific reproducibility
+- Statistical reproducibility
+
+### Why do we care about reproducibility?
+
+Take 5-10 minutes to read over the following blog post section on [reproducibility
+in science](http://bioconnector.org/workshops/r-rmarkdown.html#who_cares_about_reproducible_research)
+and [what's in it for you](http://bioconnector.org/workshops/r-rmarkdown.html#what’s_in_it_for_you).
+
+Now, let's discuss the following questions:
+1. Why does reproducibility matter in science?
+2. What do you think about when you hear the term "open science"?
+3. How does open science translate back to the issue of reproducibility
+    - How does it affect collaboration, and the progress of science?
+
+### What are ways we can make research more reproducible?
+
+- Learning from software engineers who have used continuous integration to keep
+track of their projects in an automated fashion (helping to prevent from human error)
+for years. This includes the following:
+    - analyses/models are run as scripts that can be replicated on other machines
+    and environments
+    - code is version-controlled (e.g. Git/Github)
+    - code is well tested
+    - for a take on "Continuous Analysis", read this [article from ELife](https://elifesciences.org/labs/e623676c/reproducibility-automated)
+- For more recommendations on reproducible research, especially as it pertains
+to coding practices, read more [here](http://bioconnector.org/workshops/r-rmarkdown.html#some_recommendations_for_reproducible_research)
+
+### What are the barriers to reproducibility?
+
+- Brainstorm reasons why scientists are not readily embracing reproducible science.
+
+```{r, echo=FALSE}
+#Some ideas:
+#    - workflows are hard to document at every step
+#    - we already have systems in place that work, and would be hard/time consuming to upend them
+#    - people and skills, currently low adoption and hard to force collaborators to change
```

Comment:
- Older PIs not willing to change and say they do not have time to relearn.
- Many students doing this might in a lab where the PI is not computational and does not fully appreciate the benefits. It is also hard to take the first steps as a student not knowing what the state of the art is.
- Few, if any, courses in this at the university, especially outside of computer science. Imo, courses in best practice in these areas should replace the Matlab seminars that I keep getting emails about, where they essentially just try to pitch their products to student who are better of learning an open language instead... But it is difficult when since there are no Open source reps in the same sense that there are company reps wanting to sell stuff /endrant